src/resources/defi/utils.ts

```diff
@@ -3,87 +3,181 @@ import {
   AddysPositionsResponse,
   Borrow,
   Claimable,
+  Deposit,
   Position,
   PositionsTotals,
   RainbowBorrow,
   RainbowClaimable,
   RainbowDeposit,
   RainbowPosition,
   RainbowPositions,
+  RainbowStake,
+  Stake,
+  UnderlyingAsset,
 } from './types';
-import { add, convertAmountToNativeDisplay, convertRawAmountToNativeDisplay, subtract } from '@/helpers/utilities';
+import { add, convertAmountToNativeDisplay, convertRawAmountToNativeDisplay, lessThan, subtract } from '@/helpers/utilities';
 import { maybeSignUri } from '@/handlers/imgix';
 import { ethereumUtils } from '@/utils';
 import { chainsIdByName } from '@/chains';
 
-export const parsePosition = (position: Position, currency: NativeCurrencyKey): RainbowPosition => {
-  let totalLocked = '0';
+const PROTOCOL_VERSION_REGEX = /-[vV]\d+$/;
+const LP_POOL_SYMBOL = 'LP-POOL';
+const CONCENTRATED_LIQUIDITY_ONLY_DAPPS = ['uniswap-v3'];
 
-  let totalDeposits = '0';
-  const parsedDeposits = position.deposits?.map((deposit): RainbowDeposit => {
+// Zerion currently has no specific lp position signifier
+function isLpPositionItem(item: Deposit | Stake): boolean {
+  return item.asset.symbol === LP_POOL_SYMBOL;
+}
+
+function calculatePositionItemNativeDisplayValue(item: UnderlyingAsset | Claimable | Stake, currency: NativeCurrencyKey) {
+  const decimals = typeof item.asset.decimals === 'number' ? item.asset.decimals : 18;
+  return convertRawAmountToNativeDisplay(item.quantity, decimals, item.asset.price?.value!, currency);
```

Comment:
instead of an assertion, can we use nullish coalescence? Or maybe exit early if no price value?
```
item.asset.price?.value ?? '0'
````

Reply:
This code isn't new I just wrapped it in a function. Agreed that would be cleaner, but the commit related to this code mentioned something about fixing zero decimal assets so didn't want to break it if it was like that for a reason.

Reply:
Will add the early exit if no price value though, that should be safest.